### PR TITLE
Set up Daffio template and app routing

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: apps/daffio/src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,31 +1,30 @@
 import { TestBed, async } from '@angular/core/testing';
 import { AppComponent } from './app.component';
-import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
-import { By } from '@angular/platform-browser';
+import { CUSTOM_ELEMENTS_SCHEMA, Component } from '@angular/core';
+import { RouterTestingModule } from '@angular/router/testing';
+
+@Component({selector: 'homepage', template: ''})
```
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/daffio/src/app/app.component.spec.ts]: I used foundation-demo's app files as reference and this component was included.

File: apps/daffio/src/app/app.component.spec.ts [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
       schemas: [
         CUSTOM_ELEMENTS_SCHEMA
       ],
+      imports: [
+        RouterTestingModule
+      ],
       declarations: [
-        AppComponent
+        AppComponent,
+        HomepageContainerMock
```
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/daffio/src/app/app.component.spec.ts]: I used foundation-demo's app files as reference and this mock was included.

File: apps/daffio/src/app/app.component.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (damienwebdev) [reply apps/daffio/src/app/app.component.spec.ts]: Do we need it? Is it used in the test?

File: apps/daffio/src/app/app-routing.module.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { NgModule } from '@angular/core';
+import { TemplateComponent } from './core//template/template.component';
+
+export const appRoutes: Routes = [
+  {
+    path: '', component: TemplateComponent, children: [
+      { path: '', pathMatch: 'full', loadChildren: './content/homepage/homepage.module#DaffioHomepageModule'}
+    ]
+  },
+  {
```
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/daffio/src/app/app-routing.module.ts]: Resolved in d89d323

File: apps/daffio/src/app/app.component.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { Component } from '@angular/core';
 
 @Component({
-  selector: 'daffodil-root',
+  selector: 'daffio-root',
```
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/daffio/src/app/app.component.ts]: resolved in 6c1c43f

File: apps/daffio/src/index.html [hunk 5]
```diff
@@ -10,6 +10,6 @@
   <link rel="icon" type="image/x-icon" href="favicon.ico">
 </head>
 <body>
-  <daffodil-root></daffodil-root>
+  <daffio-root></daffio-root>
```
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/daffio/src/index.html]: resolved in 6c1c43f

File: apps/daffio/src/app/app.component.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/daffio/src/app/app.component.spec.ts]: resolved in 24c4c9d

File: apps/daffio/src/app/app.component.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/daffio/src/app/app.component.spec.ts]: resolved in 24c4c9d

File: apps/daffio/src/app/core/template/template.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  });
+
+  it('should render a `daffio-header-container` inside a `sidebar-viewport-container`', () => {
+    const fixture = TestBed.createComponent(TemplateComponent);
+
+    let sidebarViewport = fixture.debugElement.query(By.css('sidebar-viewport-container'));
+    
+    expect(sidebarViewport.query(By.css('daffio-header-container'))).not.toBeNull()
+  })
+});
```
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/daffio/src/app/core/template/template.component.spec.ts]: resolved in 24c4c9d

File: apps/daffio/src/app/app.component.spec.ts [hunk 7]
```diff
@@ -1,31 +1,30 @@
...
-  });
-
-  it('should render a `daffio-header-container` inside a `sidebar-viewport-container`', () => {
-    const fixture = TestBed.createComponent(AppComponent);
-
-    let sidebarViewport = fixture.debugElement.query(By.css('sidebar-viewport-container'));
-    
-    expect(sidebarViewport.query(By.css('daffio-header-container'))).not.toBeNull()
-  })
+  }));
```
[the remark this answers is not in the document]
Comment (xelaint) [reply apps/daffio/src/app/app.component.spec.ts]: resolved in 24c4c9d

**Pull request merged**